[BUG] Sphinx 7.2.6 raises ThemeError

### Software versions

Python version        :  3.10.13 | packaged by conda-forge | (main, Oct 26 2023, 18:07:37) [GCC 12.3.0]
IPython version       :  8.18.0
Tornado version       :  6.3.3
Bokeh version         :  3.4.0.dev3+27.g2555a4f2.dirty
BokehJS static path   :  /home/jovyan/private/bokeh/bokehjs/build
node.js version       :  v20.9.0
npm version           :  10.2.4
jupyter_bokeh version :  (not installed)
Operating system      :  Linux-4.19.0-21-amd64-x86_64-with-glibc2.35

### Observed behavior

I tried to build the docs with the latest version of Sphinx after creating a new `bkdev` conda environment.

```
Sphinx==7.2.6
sphinx-copybutton==0.5.2
sphinx-favicon==1.0.1
sphinx_design==0.5.0
sphinxcontrib-applehelp==1.0.7
sphinxcontrib-devhelp==1.0.5
sphinxcontrib-htmlhelp==2.0.4
sphinxcontrib-jsmath==1.0.1
sphinxcontrib-qthelp==1.0.6
sphinxcontrib-serializinghtml==1.1.9
```

It looks like the Sphinx version is not pinned to <7.x and therefor I got the latest version of the package. Running `make html` raise the error below.

```
Theme error:
An error happened in rendering the page docs/dev_guide.
Reason: ThemeError("Local asset file paths must not contain query strings: '_static/custom.css?v=3.4.0.dev3+27.g2555a4f2.dirty'")
```

This is because of [line 167 in `conf.py`](
https://github.com/bokeh/bokeh/blame/2555a4f2c91c3096b8e7a669a51d727d403b031b/docs/bokeh/source/conf.py#L167)

```
html_css_files = [f"custom.css?v={version}"]
```

which should be changed to 

```
html_css_files = ["custom.css"]
```
.

This will become more important if the next documentation should be build with the latest version of sphinx in a future release.

I'm also having an error when trying to build the docs locally, although it's of a different type:

```
Running Sphinx v7.2.6
/Users/mac/Documents/development/bokeh/src/bokeh/sphinxext/bokeh_gallery.py:44: RemovedInSphinx80Warning: The alias 'sphinx.util.status_iterator' is deprecated, use 'sphinx.util.display.status_iterator' instead. Check CHANGES for Sphinx API modifications.
  from sphinx.util import ensuredir, status_iterator
/Users/mac/Documents/development/bokeh/src/bokeh/sphinxext/bokeh_plot.py:104: RemovedInSphinx80Warning: The alias 'sphinx.util.status_iterator' is deprecated, use 'sphinx.util.display.status_iterator' instead. Check CHANGES for Sphinx API modifications.
  from sphinx.util import copyfile, ensuredir, status_iterator
/Users/mac/Documents/development/bokeh/src/bokeh/sphinxext/bokeh_sitemap.py:41: RemovedInSphinx80Warning: The alias 'sphinx.util.status_iterator' is deprecated, use 'sphinx.util.display.status_iterator' instead. Check CHANGES for Sphinx API modifications.
  from sphinx.util import status_iterator
creating gallery file entries... [100%] missing_dates.rst
[autosummary] generating autosummary for: docs/dev_guide.rst, docs/dev_guide/bindings.rst, docs/dev_guide/bokehjs.rst, docs/dev_guide/creating_issues.rst, docs/dev_guide/debug_headless_chrome.rst, docs/dev_guide/documentation.rst, docs/dev_guide/env_vars.rst, docs/dev_guide/models.rst, docs/dev_guide/pull_requests.rst, docs/dev_guide/python.rst, ..., docs/user_guide/topics/contour.rst, docs/user_guide/topics/geo.rst, docs/user_guide/topics/graph.rst, docs/user_guide/topics/hex.rst, docs/user_guide/topics/hierarchical.rst, docs/user_guide/topics/images.rst, docs/user_guide/topics/pie.rst, docs/user_guide/topics/stats.rst, docs/user_guide/topics/timeseries.rst, index.rst
BokehDeprecationWarning: 'tile_providers module' was deprecated in Bokeh 3.0.0 and will be removed, use 'add_tile directly' instead.
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://pandas.pydata.org/pandas-docs/stable/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://www.sphinx-doc.org/en/master/objects.inv...
loading intersphinx inventory from https://xyzservices.readthedocs.io/en/stable/objects.inv...
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 607 source files that are out of date
updating environment: [new config] 607 added, 0 changed, 0 removed
Compilation failed:[  2%] docs/examples/advanced/extensions/putting_together

../../examples/advanced/extensions/putting_together.py:Custom.ts:3:22 - error TS2307: Cannot find module 'models/widgets/sliders/slider' or its corresponding type declarations.

3 import {Slider} from "models/widgets/sliders/slider"
                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make: *** [html] Error 1
```

> 
> Can you check if you have installed `bokehjs` correctly. On Saturday I was able to build the docs after I changed the one line in the `conf.py` as described above. I expect that it should work for you, too.

Thanks. So, I rebuilt my `bokehjs` dependencies and re-ran `pip install -e .`  However, after running `make clean html serve`, I got a different error message(something to do with the now deprecated Circle glyph):

```
rm -rf build/*
rm -rf source/docs/gallery/*
rm -rf source/docs/examples/*
Running Sphinx v7.2.6
/Users/mac/Documents/development/bokeh/src/bokeh/sphinxext/bokeh_gallery.py:44: RemovedInSphinx80Warning: The alias 'sphinx.util.status_iterator' is deprecated, use 'sphinx.util.display.status_iterator' instead. Check CHANGES for Sphinx API modifications.
  from sphinx.util import ensuredir, status_iterator
/Users/mac/Documents/development/bokeh/src/bokeh/sphinxext/bokeh_plot.py:104: RemovedInSphinx80Warning: The alias 'sphinx.util.status_iterator' is deprecated, use 'sphinx.util.display.status_iterator' instead. Check CHANGES for Sphinx API modifications.
  from sphinx.util import copyfile, ensuredir, status_iterator
/Users/mac/Documents/development/bokeh/src/bokeh/sphinxext/bokeh_sitemap.py:41: RemovedInSphinx80Warning: The alias 'sphinx.util.status_iterator' is deprecated, use 'sphinx.util.display.status_iterator' instead. Check CHANGES for Sphinx API modifications.
  from sphinx.util import status_iterator
making output directory... done
creating gallery file entries... [100%] missing_dates.rst
[autosummary] generating autosummary for: docs/dev_guide.rst, docs/dev_guide/bindings.rst, docs/dev_guide/bokehjs.rst, docs/dev_guide/creating_issues.rst, docs/dev_guide/debug_headless_chrome.rst, docs/dev_guide/documentation.rst, docs/dev_guide/env_vars.rst, docs/dev_guide/models.rst, docs/dev_guide/pull_requests.rst, docs/dev_guide/python.rst, ..., docs/user_guide/topics/contour.rst, docs/user_guide/topics/geo.rst, docs/user_guide/topics/graph.rst, docs/user_guide/topics/hex.rst, docs/user_guide/topics/hierarchical.rst, docs/user_guide/topics/images.rst, docs/user_guide/topics/pie.rst, docs/user_guide/topics/stats.rst, docs/user_guide/topics/timeseries.rst, index.rst
BokehDeprecationWarning: 'tile_providers module' was deprecated in Bokeh 3.0.0 and will be removed, use 'add_tile directly' instead.
loading intersphinx inventory from https://numpy.org/doc/stable/objects.inv...
loading intersphinx inventory from https://pandas.pydata.org/pandas-docs/stable/objects.inv...
loading intersphinx inventory from https://docs.python.org/3/objects.inv...
loading intersphinx inventory from https://www.sphinx-doc.org/en/master/objects.inv...
loading intersphinx inventory from https://xyzservices.readthedocs.io/en/stable/objects.inv...
building [mo]: targets for 0 po files that are out of date
writing output... 
building [html]: targets for 607 source files that are out of date
updating environment: [new config] 607 added, 0 changed, 0 removed
/Users/mac/Documents/development/bokeh/examples/basic/annotations/band.py:23: FutureWarning: The provided callable <function mean at 0x1038d7a60> is currently using Rolling.mean. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "mean" instead.
  df2 = df.y.rolling(window=300).agg({"y_mean": np.mean, "y_std": np.std})
/Users/mac/Documents/development/bokeh/examples/basic/annotations/band.py:23: FutureWarning: The provided callable <function std at 0x1038d7ba0> is currently using Rolling.std. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "std" instead.
  df2 = df.y.rolling(window=300).agg({"y_mean": np.mean, "y_std": np.std})
WARNING:bokeh.embed.util: docs/examples/interaction/js_callbacks/doc_js_events
You are generating standalone HTML/JS output, but trying to use real Python
callbacks (i.e. with on_change or on_event). This combination cannot work.

Only JavaScript callbacks may be used with standalone output. For more
information on JavaScript callbacks with Bokeh, see:

    https://docs.bokeh.org/en/latest/docs/user_guide/interaction/callbacks.html

Alternatively, to use real Python callbacks, a Bokeh server application may
be used. For more information on building and running Bokeh applications, see:

    https://docs.bokeh.org/en/latest/docs/user_guide/server.html

reading sources... [ 36%] docs/examples/models/glyphs
ERROR:bokeh.core.validation.check:E-1001 (BAD_COLUMN_NAME): Glyph refers to nonexistent column name. This could either be due to a misspelling or typo, or due to an expected column being missing. : text_font='Helvetica' [no close matches] {renderer: GlyphRenderer(id='p30887', ...)}
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'triangle() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='triangle', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'triangle() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='triangle', ...) instead" instead.
BokehDeprecationWarning: 'square() method' was deprecated in Bokeh 3.3.0 and will be removed, use "scatter(marker='square', ...) instead" instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
BokehDeprecationWarning: 'circle() method with size value' was deprecated in Bokeh 3.3.0 and will be removed, use 'scatter(size=...) instead' instead.
<bokeh-content>:1: ERROR: Unknown target name: "multiple kernel density estimation".
<bokeh-prop>:1: ERROR: Unexpected indentation.
<bokeh-model>:1: WARNING: Block quote ends without a blank line; unexpected unindent.
<bokeh-prop>:1: ERROR: Unexpected indentation.
<bokeh-model>:1: WARNING: Block quote ends without a blank line; unexpected unindent.
copying bokeh-plot files... [100%] bokeh-content-fe9eb6a78d684a5db9fb244c9174eaf7-docs-examples-plotting-histogram.js
adding links to sitemap... 

Sphinx error:
Error generating bokeh-content-40362131fa9044b5b90617c7c8a6f5b7-docs-reference-models-glyphs-circle.js: 

bokeh-plot:: error:

Traceback (most recent call last):
  File "/Users/mac/Documents/development/bokeh/src/bokeh/application/handlers/code_runner.py", line 229, in run
    exec(self._code, module.__dict__)
  File "/Users/mac/Documents/development/bokeh/examples/reference/models/Circle.py", line 17, in <module>
    glyph = Circle(x="x", y="y", size="sizes", line_color="#3288bd", fill_color="white", line_width=3)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mac/Documents/development/bokeh/src/bokeh/models/glyphs.py", line 452, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/models/glyph.py", line 80, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/models/glyph.py", line 101, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/models/glyph.py", line 111, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/models/glyph.py", line 131, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/models/glyph.py", line 58, in __init__
    super().__init__(*args, **kwargs)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/model/model.py", line 110, in __init__
    super().__init__(**kwargs)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/core/has_props.py", line 305, in __init__
    setattr(self, name, value)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/core/has_props.py", line 343, in __setattr__
    self._raise_attribute_error_with_matches(name, properties)
  File "/Users/mac/Documents/development/bokeh/src/bokeh/core/has_props.py", line 378, in _raise_attribute_error_with_matches
    raise AttributeError(f"unexpected attribute {name!r} to {self.__class__.__name__}, {text} attributes are {nice_join(matches)}")
AttributeError: unexpected attribute 'size' to Circle, possible attributes are decorations, fill_alpha, fill_color, hatch_alpha, hatch_color, hatch_extra, hatch_pattern, hatch_scale, hatch_weight, hit_dilation, js_event_callbacks, js_property_callbacks, line_alpha, line_cap, line_color, line_dash, line_dash_offset, line_join, line_width, name, radius, radius_dimension, radius_units, subscribed_events, syncable, tags, x or y


evaluating source:

import numpy as np

from bokeh.io import curdoc, show
from bokeh.models import Circle, ColumnDataSource, Grid, LinearAxis, Plot

N = 9
x = np.linspace(-2, 2, N)
y = x**2
sizes = np.linspace(10, 20, N)

source = ColumnDataSource(dict(x=x, y=y, sizes=sizes))

plot = Plot(
    title=None, width=300, height=300,
    min_border=0, toolbar_location=None)

glyph = Circle(x="x", y="y", size="sizes", line_color="#3288bd", fill_color="white", line_width=3)
plot.add_glyph(source, glyph)

xaxis = LinearAxis()
plot.add_layout(xaxis, 'below')

yaxis = LinearAxis()
plot.add_layout(yaxis, 'left')

plot.add_layout(Grid(dimension=0, ticker=xaxis.ticker))
plot.add_layout(Grid(dimension=1, ticker=yaxis.ticker))

curdoc().add_root(plot)

show(plot)

make: *** [html] Error 2
```

Progress, if you ask me :)

--- a/conda/environment-release-build.yml
+++ b/conda/environment-release-build.yml
@@ -38,7 +38,7 @@ dependencies:
   - pip:
     # docs
     - pydata_sphinx_theme
-    - sphinx < 7
+    - sphinx >= 7.1.0
     - sphinx-copybutton
     - sphinx-design
     - sphinx-favicon

--- a/conda/environment-test-3.10.yml
+++ b/conda/environment-test-3.10.yml
@@ -68,7 +68,7 @@ dependencies:
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
-    - sphinx
+    - sphinx >= 7.1.0
     - sphinx-copybutton
     - sphinx-design
     - sphinx-favicon

--- a/conda/environment-test-3.11.yml
+++ b/conda/environment-test-3.11.yml
@@ -68,7 +68,7 @@ dependencies:
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
-    - sphinx
+    - sphinx >= 7.1.0
     - sphinx-copybutton
     - sphinx-design
     - sphinx-favicon

--- a/conda/environment-test-3.12.yml
+++ b/conda/environment-test-3.12.yml
@@ -68,7 +68,7 @@ dependencies:
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
-    - sphinx
+    - sphinx >= 7.1.0
     - sphinx-copybutton
     - sphinx-design
     - sphinx-favicon

--- a/conda/environment-test-3.9.yml
+++ b/conda/environment-test-3.9.yml
@@ -68,7 +68,7 @@ dependencies:
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
-    - sphinx
+    - sphinx >= 7.1.0
     - sphinx-copybutton
     - sphinx-design
     - sphinx-favicon

--- a/conda/environment-test-minimal-deps.yml
+++ b/conda/environment-test-minimal-deps.yml
@@ -57,7 +57,7 @@ dependencies:
     # docs
     - pydata_sphinx_theme
     - requests-unixsocket >= 0.3.0
-    - sphinx
+    - sphinx >= 7.1.0
     - sphinx-copybutton
     - sphinx-design
     - sphinx-favicon

--- a/docs/bokeh/source/conf.py
+++ b/docs/bokeh/source/conf.py
@@ -164,7 +164,7 @@ html_context = {
     "VERSION": version,
 }
 
-html_css_files = [f"custom.css?v={version}"]
+html_css_files = ["custom.css"]
 
 html_static_path = ["_static"]
 

